src/icc-tags.js

```diff
@@ -0,0 +1,150 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+import {iccTags, iccProfile} from './icc-tag-names';
+import {getStringFromDataView, getUnicodeStringFromDataView} from './utils';
+
+export default {
+    read
+};
+
+const PROFILE_HEADER_LENGTH = 84;
+
+function read(dataView, iccData) {
+    try {
+        const totalIccProfileLength = iccData.reduce((sum, icc) => sum + icc.length, 0);
+
+        const iccBinaryData = new Uint8Array(totalIccProfileLength);
+        let offset = 0;
+        const buffer = dataView.buffer;
+        for (let chunkNumber = 1; chunkNumber <= iccData.length; chunkNumber++) {
+            const iccDataChunk = iccData.find((x) => x.chunkNumber === chunkNumber);
+            if (!iccDataChunk) {
+                throw new Error(`ICC chunk ${chunkNumber} not found`);
+            }
+
+            const data = buffer.slice(iccDataChunk.offset, iccDataChunk.offset + iccDataChunk.length);
+            const chunkData = new Uint8Array(data);
+
+            iccBinaryData.set(chunkData, offset);
+            offset += chunkData.length;
+        }
+
+        return parseTags(new DataView(iccBinaryData.buffer));
+    } catch (error) {
+        return {};
+    }
+}
+
+export function parseTags(dataView) {
```

Comment:
This comment from the last review disappeared: This function has a lot of magic numbers. Could they get their own constants with some descriptive names?

Reply:
with tag types now being named, I am not sure that introducing more constants will help readability. Variable names should already be descriptive enough. E.g.: ```
 const countryCode = getStringFromDataView(dataView, offset + 2, 2);
```
I really don't see the point of having another COUNTRY_CODE_OFFSET - if somebody wants to update the code, they will have ICC spec open and will be looking at offsets directly. Going to other part of the file to see/edit actual number would just take longer - or maybe that is just me.

Reply:
Good point, definitely agree on `countryCode` and the likes, and I won't fight you on the other ones. :-)